internal/util/util.go

```diff
@@ -107,23 +108,32 @@ func CheckExistingBundlesMatchesTemplate(existingBundles *rukpakv1alpha1.BundleL
 }
 
 // CheckDesiredBundleTemplate is responsible for determining whether the existingBundle
-// parameter is semantically equal to the desiredBundle Bundle template.
+// hash is equal to the desiredBundle Bundle template hash.
 func CheckDesiredBundleTemplate(existingBundle *rukpakv1alpha1.Bundle, desiredBundle *rukpakv1alpha1.BundleTemplate) bool {
-	// Create a copy of the existingBundle resource and delete the known set of
-	// auto-generated labels that the BI controller injects when comparing the
-	// current and desired set of labels.
-	bundle := existingBundle.DeepCopy()
-	delete(bundle.Labels, CoreOwnerKindKey)
-	delete(bundle.Labels, CoreOwnerNameKey)
-	return equality.Semantic.DeepEqual(bundle.Spec, desiredBundle.Spec) &&
-		equality.Semantic.DeepEqual(existingBundle.Annotations, desiredBundle.Annotations) &&
-		equality.Semantic.DeepEqual(bundle.Labels, desiredBundle.Labels)
-}
-
-// GenerateBundleName is responsible for generating a unique
-// Bundle metadata.Name using the biName parameter as the base.
-func GenerateBundleName(biName string) string {
-	return fmt.Sprintf("%s-%s", biName, rand.String(5))
+	if len(existingBundle.Labels) == 0 {
+		// Existing Bundle has no labels set, which should never be the case.
+		// Return false so that the Bundle is forced to be recreated with the expected labels.
+		return false
+	}
+
+	var existingHash string
+	if hash, ok := existingBundle.Labels[CoreBundleTemplateHashKey]; ok {
+		existingHash = hash
+	} else {
+		// Existing Bundle has no template hash associated with it.
+		// Return false so that the Bundle is forced to be recreated with the template hash label.
+		return false
+	}
```

Comment:
super nit: flip this conditional to !ok so we can exit early and remove the if/else conditional structure?